src/v2/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -146,211 +141,180 @@ export class ShippingRoute extends Component<ShippingProps, ShippingState> {
             shipping: shipToAddress,
             phoneNumber: shipToPhoneNumber,
           },
         })
       ).commerceSetShipping?.orderOrError
 
       if (orderOrError?.error) {
-        this.handleSubmitError(orderOrError.error)
+        handleSubmitError(orderOrError.error)
         return
       }
       // save address when user is entering new address AND save checkbox is selected
-      await this.saveAddress()
+      await saveAddress()
 
-      if (isArtaShipping) {
-        this.setState({
-          shippingQuotes: getShippingQuotes(orderOrError?.order),
-        })
+      if (isArtaShipping()) {
+        console.log({ shippingQuotes: getShippingQuotes(orderOrError?.order) })
+        setShippingQuotes(getShippingQuotes(orderOrError?.order))
       } else {
-        this.props.router.push(`/orders/${this.props.order.internalID}/payment`)
+        props.router.push(`/orders/${props.order.internalID}/payment`)
       }
     } catch (error) {
       logger.error(error)
-      this.props.dialog.showErrorDialog()
+      props.dialog.showErrorDialog()
     }
   }
 
-  selectShippingQuote = async () => {
-    const { shippingQuoteId } = this.state
-    const { order } = this.props
-
-    if (shippingQuoteId && order.internalID) {
+  const selectShippingQuote = async () => {
+    if (shippingQuoteId && props.order.internalID) {
       try {
         const orderOrError = (
-          await selectShippingOption(this.props.commitMutation, {
+          await selectShippingOption(props.commitMutation, {
             input: {
-              id: order.internalID,
+              id: props.order.internalID,
               selectedShippingQuoteId: shippingQuoteId,
             },
           })
         ).commerceSelectShippingOption?.orderOrError
 
-        await this.saveAddress()
+        await saveAddress()
 
         if (orderOrError?.error) {
-          this.handleSubmitError(orderOrError.error)
+          handleSubmitError(orderOrError.error)
           return
         }
 
-        this.props.router.push(`/orders/${this.props.order.internalID}/payment`)
+        props.router.push(`/orders/${props.order.internalID}/payment`)
       } catch (error) {
         logger.error(error)
-        this.props.dialog.showErrorDialog({
-          message: this.getArtaErrorMessage(),
+        props.dialog.showErrorDialog({
+          message: getArtaErrorMessage(),
         })
       }
     }
   }
 
-  saveAddress = async () => {
-    const {
-      address,
-      phoneNumber,
-      shippingOption,
-      saveAddress,
-      savedAddressID,
-    } = this.state
-    const { relayEnvironment } = this.props
-    const isCreateNewAddress = this.isCreateNewAddress()
-
-    if (saveAddress) {
-      if (shippingOption === "SHIP" && isCreateNewAddress && relayEnvironment) {
+  const saveAddress = async () => {
+    if (saveAddressToggle) {
+      if (
+        shippingOption === "SHIP" &&
+        isCreateNewAddress() &&
+        props.relayEnvironment
```

Comment:
Is the removal of the destructuring of `relayEnvironment` from `props` intentional? I don't personally care either way but just noting that we do end up with `props.` sprinkled into this function multiple times to access `relayEnvironment`.

Reply:
Good call. Looking into this I realized there was also a [remaining destructuring from the older render() function](https://github.com/artsy/force/pull/8918/files#diff-3450e2d03077be9b8efde9cbfc7e452910188d0d80ad0cd4848b242d63840b60R511).
I'm destructuring all props together now 👌